src/v/security/oidc_service.cc

```diff
@@ -19,67 +23,114 @@
 #include <seastar/core/lowres_clock.hh>
 #include <seastar/core/sstring.hh>
 #include <seastar/coroutine/as_future.hh>
+#include <seastar/coroutine/exception.hh>
+#include <seastar/util/defer.hh>
 
 #include <absl/algorithm/container.h>
+#include <absl/container/flat_hash_map.h>
+#include <boost/outcome/success_failure.hpp>
+
+#include <chrono>
 
 namespace security::oidc {
 
 namespace {
-ss::future<ss::sstring> make_request(std::string_view url_view) {
-    auto url_res = security::oidc::parse_url(url_view);
-    if (url_res.has_error()) {
-        throw std::runtime_error("Invalid url");
+
+template<typename... Args>
+[[nodiscard]] auto return_exception(
+  std::error_code ec,
+  fmt::format_string<Args...> fmt,
+  Args&&... args) noexcept {
+    return ss::coroutine::return_exception(
+      exception(ec, fmt::format(fmt, std::forward<Args>(args)...)));
+}
+
+using seastar::operator co_await;
+
+class probe {
+public:
+    using hist_t = log_hist_public;
+
+    class auto_measure {
+    public:
+        explicit auto_measure(probe* p)
+          : _p{p}
+          , _m{p->_request_latency} {}
+
+    public:
+        void success() noexcept {
+            if (std::exchange(_p, nullptr) != nullptr) {
+                auto report_and_destroy = std::move(_m);
```

Comment:
What's to prevent this move assignment from getting blown away by an optimization pass?

Reply:
> What's to prevent this move assignment from getting blown away by an optimization pass?

Whilst copy and move constructors are allowed to be elided in some cases, this is not one of them. The destructor must be called at the end of this scope, or am I missing something?

Reply:
No, that sounds right. Thanks